src/source/workspace.ml

```diff
@@ -564,7 +564,9 @@ module Context = struct
     let decode =
       let+ common = Common.decode
       and+ name =
-        field_o "name" (Dune_lang.Syntax.since syntax (1, 10) >>> Context_name.decode)
+        field_o
+          "name"
+          (located (Dune_lang.Syntax.since syntax (1, 10) >>> Context_name.decode))
```

Comment:
I do not understand this change at all. If you introduce a location here and then drop it in the bottom you could just... not do that?